packages/connectivity_plus/tizen/src/connection.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "connection.h"
+
+#include "log.h"
+
+static ConnectionType ToConnectionType(connection_type_e type) {
+  switch (type) {
+    case CONNECTION_TYPE_WIFI:
+      return ConnectionType::kWiFi;
+    case CONNECTION_TYPE_CELLULAR:
+      return ConnectionType::kMobile;
+    case CONNECTION_TYPE_ETHERNET:
+      return ConnectionType::kEthernet;
+    case CONNECTION_TYPE_DISCONNECTED:
+    default:
+      return ConnectionType::kNone;
+  }
+}
+
+Connection::Connection() {
+  int ret = connection_create(&connection_);
+  if (ret != CONNECTION_ERROR_NONE) {
+    LOG_ERROR("Failed to create handle: %s", get_error_message(ret));
+    last_error_ = ret;
+  }
+}
+
+Connection::~Connection() {
+  if (connection_) {
+    connection_destroy(connection_);
```

Comment:
For stability, it is recommended to unset the callback before destroying the connection handle.

Reply:
Fixed!